shared/desktop/webpack.config.babel.js

```diff
@@ -96,222 +72,94 @@ const makeCommonConfig = () => {
         use: ['style-loader', 'css-loader'],
       },
     ]
-  }
 
-  const makeCommonPlugins = () => {
+    // If we use the hot server it pulls in this config
+    const devServer = {
+      compress: false,
+      contentBase: path.resolve(__dirname, 'dist'),
+      hot: isHot,
+      lazy: false,
+      overlay: true,
+      port: 4000,
+      publicPath: 'http://localhost:4000/dist/',
+      quiet: false,
+      stats: {colors: true},
+    }
+
     const defines = {
-      __DEV__: flags.isDev,
-      __HOT__: flags.isHot,
-      __SCREENSHOT__: flags.isVisDiff,
+      __DEV__: isDev,
+      __HOT__: isHot,
+      __SCREENSHOT__: false,
       __STORYBOOK__: false,
-      __VERSION__: flags.isDev ? JSON.stringify('Development') : JSON.stringify(process.env.APP_VERSION),
-      'process.env.NODE_ENV': flags.isDev ? JSON.stringify('development') : JSON.stringify('production'),
+      __VERSION__: isDev ? JSON.stringify('Development') : JSON.stringify(process.env.APP_VERSION),
     }
     console.warn('Injecting defines: ', defines)
-    const definePlugin = [new webpack.DefinePlugin(defines)]
 
-    const uglifyPlugin = flags.isDev
-      ? []
-      : [
-          new UglifyJSPlugin({
-            sourceMap: true,
-            uglifyOptions: {
-              output: {
-                comments: false,
-              },
+    return {
+      bail: true,
+      devServer,
+      mode: isDev ? 'development' : 'production',
+      module: {rules},
+      node: {__dirname: true},
+      output: {
+        filename: '[name].bundle.js',
+        path: path.resolve(__dirname, 'dist'),
+        publicPath: isHot ? 'http://localhost:4000/dist/' : '../dist/',
+      },
+      plugins: [
+        new webpack.DefinePlugin(defines), // Inject some defines
+        new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/), // Skip a bunch of crap moment pulls in
+      ],
+      resolve: {
+        extensions: ['.desktop.js', '.js', '.jsx', '.json', '.flow'],
+      },
+      stats: {
+        optimizationBailout: true,
+      },
+      ...(isDev
+        ? {}
+        : {
+            optimization: {
+              minimizer: [
+                new UglifyJSPlugin({
```

Comment:
by default in production mode it'll actually have uglifyjs but it doesn't (i think) let us override the default settings so we just run this ourselves